src/URI.js

```diff
@@ -1241,10 +1241,14 @@
     } else if (_URI || _object) {
       var src = _URI ? href._parts : href;
       for (key in src) {
+        if (key === 'query') {
+          continue;
+        }
         if (hasOwn.call(this._parts, key)) {
           this._parts[key] = src[key];
         }
       }
+      this.query(src['query']);
```

Comment:
this should probably only be executed if `src.query` is actually defined

Reply:
definitely! will update.